core/server/src/main/java/alluxio/worker/block/evictor/LIRSEvictor.java

```diff
@@ -0,0 +1,573 @@
+   * called.
+   */
+  private Map<Long, Long> mBlockIdToSize = new ConcurrentHashMap<Long, Long>();
+  /**
+   * Record the information of all LIR blocks. Pair<BlockStoreLocation, Long> identifies a block
+   * because a block may contain more than one ghost blocks on all tiers.
+   */
+  private Map<Pair<BlockStoreLocation, Long>, BlockLIRSInfo> mLIRCache =
+      Collections.synchronizedMap(new LinkedHashMap<Pair<BlockStoreLocation, Long>, BlockLIRSInfo>(
+          LINKED_HASH_MAP_INIT_CAPACITY, LINKED_HASH_MAP_INIT_LOAD_FACTOR,
+          LINKED_HASH_MAP_ACCESS_ORDERED));
+  // Record the information of all HIR blocks.
+  private Map<Pair<BlockStoreLocation, Long>, BlockLIRSInfo> mHIRCache =
+      Collections.synchronizedMap(new LinkedHashMap<Pair<BlockStoreLocation, Long>, BlockLIRSInfo>(
+          LINKED_HASH_MAP_INIT_CAPACITY, LINKED_HASH_MAP_INIT_LOAD_FACTOR,
+          LINKED_HASH_MAP_ACCESS_ORDERED));
+  /**
+   * Record the information of all temp blocks. Temp blocks are blocks moved from other tiers and
+   * still not accessed by any clients.
+   */
+  private Map<Pair<BlockStoreLocation, Long>, Boolean> mTmpCache = Collections.synchronizedMap(
+      new LinkedHashMap<Pair<BlockStoreLocation, Long>, Boolean>(LINKED_HASH_MAP_INIT_CAPACITY,
+          LINKED_HASH_MAP_INIT_LOAD_FACTOR, LINKED_HASH_MAP_ACCESS_ORDERED));
+
+  /**
+   * Create a new instance of {@link LIRSEvictor}.
+   *
+   * @param view a view of block metadata information
+   * @param allocator an allocation policy
+   */
+  public LIRSEvictor(BlockMetadataManagerView view, Allocator allocator) {
+    super(view, allocator);
+    mConfiguration = WorkerContext.getConf();
+    mHIRPercent = mConfiguration.getDouble(Constants.WORKER_EVICTOR_LIRS_HIR_PERCENT);
+    Preconditions.checkArgument(mHIRPercent >= 0 && mHIRPercent <= 1,
+        "HIR percent should be larger than 0 and less than 1");
+    mLIRPercent = 1.0 - mHIRPercent;
+
+    for (StorageTierView tier : view.getTierViews()) {
+      for (StorageDirView dir : tier.getDirViews()) {
+        long lirBytes = 0;
+        long hirBytes = 0;
+        long totalBytes = dir.getAvailableBytes() + dir.getEvitableBytes();
+        BlockStoreLocation location =
+            new BlockStoreLocation(tier.getTierViewAlias(), dir.getDirViewIndex());
+        for (BlockMeta blockMeta : dir.getEvictableBlocks()) {
+          long blockId = blockMeta.getBlockId();
+          long blocksize = blockMeta.getBlockSize();
+          Pair<BlockStoreLocation, Long> key =
+              new Pair<BlockStoreLocation, Long>(location, blockId);
+          // If the size of LIR blocks exceeds the limit, put it into HIR set.
+          if (lirBytes + blocksize > mLIRPercent * totalBytes) {
+            // put into HIR set
+            BlockLIRSInfo blockinfo = new BlockLIRSInfo(false, true);
+            mLIRCache.put(key, blockinfo);
+            mHIRCache.put(key, blockinfo);
+            hirBytes += blocksize;
+          } else {

```

Comment:
NIT: since blocks are put into LIR set first, this `else` block can be switched with the above `if` block, so that intuitively when reading the code straight-through, it is clearer that LIR set is to be filled first.